src/parser.cpp

```diff
@@ -36,4 +37,118 @@ types::VPDMapVariant Parser::parse()
     return l_parser->parse();
 }
 
+int Parser::updateVpdKeyword(const types::WriteVpdParams& i_paramsToWriteData)
+{
+    // Update keyword's value on hardware
+    int l_bytesUpdatedOnHardware;
+    try
+    {
+        std::shared_ptr<ParserInterface> l_vpdParserInstance =
+            getVpdParserInstance();
+        l_bytesUpdatedOnHardware =
+            l_vpdParserInstance->writeKeywordOnHardware(i_paramsToWriteData);
+    }
+    catch (const std::exception& l_exception)
+    {
+        // TODO : Log PEL
+        return -1;
+    }
+
+    auto [l_fruPath, l_inventoryObjPath, l_redundantFruPath] =
+        jsonUtility::getAllPathsToUpdateKeyword(m_parsedJson, m_vpdFilePath);
+
+    // If inventory D-bus object path is present, perform update
+    if (!l_inventoryObjPath.empty())
+    {
+        types::Record l_recordName;
+        std::string l_interfaceName;
+        std::string l_propertyName;
+        types::DbusVariantType l_keywordValue;
+
+        if (const types::IpzData* l_ipzData =
+                std::get_if<types::IpzData>(&i_paramsToWriteData))
+        {
+            l_recordName = std::get<0>(*l_ipzData);
+            l_interfaceName = constants::ipzVpdInf + l_recordName;
+            l_propertyName = std::get<1>(*l_ipzData);
+
+            try
+            {
+                // Read keyword's value from hardware to write the same on
+                // D-bus.
+                std::shared_ptr<ParserInterface> l_vpdParserInstance =
+                    getVpdParserInstance();
+
+                logging::logMessage("Performing VPD read on " + m_vpdFilePath);
+
+                l_keywordValue = l_vpdParserInstance->readKeywordFromHardware(
+                    types::ReadVpdParams(
+                        std::make_tuple(l_recordName, l_propertyName)));
+            }
+            catch (const std::exception& l_exception)
+            {
+                // TODO: Log PEL
+                // Unable to read keyword's value from hardware.
+                return -1;
+            }
+        }
+        else
+        {
+            // Input parameter type provided isn't compatible to perform update.
+            return -1;
+        }
+
+        // Create D-bus object map
+        types::ObjectMap l_dbusObjMap = {std::make_pair(
+            l_inventoryObjPath,
+            types::InterfaceMap{std::make_pair(
+                l_interfaceName, types::PropertyMap{std::make_pair(
+                                     l_propertyName, l_keywordValue)})})};
+
+        // Call PIM's Notify method to perform update
+        if (!dbusUtility::callPIM(std::move(l_dbusObjMap)))
+        {
+            // Call to PIM's Notify method failed.
```

Comment:
Log message to console in case of error scenario.

Reply:
Done.